crates/jit/src/link.rs

```diff
@@ -132,6 +152,25 @@ fn apply_reloc(
     }
 }
 
+/// Sanitizes relocation information (they may point to real addresses).
+pub fn unlink_module(obj_data: &mut [u8]) -> Result<(), String> {
+    let obj = File::parse(obj_data).map_err(|_| "Unable to read obj".to_string())?;
+    for section in obj.sections() {
+        let body_offset = match section.file_range() {
+            Some((start, _)) => start as usize,
+            None => {
+                continue;
+            }
+        };
+        let body = unsafe { obj_data.as_ptr().add(body_offset) } as *const VMFunctionBody;
```

Comment:
(similar comment about unsafety as above)

Additionally this look looks basically the same as the one above, so could that be factored out? Something like a function that applies all relocations and uses a closure to calculate the reloc value.